lib/optimized/coin_flip_fix.dart

```diff
@@ -0,0 +1,233 @@
+      isCoinFlipping = true;
+
+      _maybeScheduleTask();
+    }
+  }
+
+  void _maybeScheduleTask() {
+    if (_taskNumber <= _totalTasks) {
+      // For each task, do a small chunk of work that will complete in <= 4ms:
+      // _totalFlipCount / _totalTasks = 5000 flips per task.
+      _scheduler.scheduleTask(() => _flipNTimes(5000), Priority.animation);
+    } else {
+      // At this point, all of the tasks have been completed. Update the UI.
+      setState(() {
+        _taskNumber = 1;
+        isCoinFlipping = false;
+      });
+    }
+  }
+
+  void _flipNTimes(int n) {
+    for (int i = 0; i < n; i++) {
+      bool isHeads = random.nextInt(2) % 2 == 0;
+      if (isHeads) {
+        _headsCount++;
+      } else {
+        _tailsCount++;
+      }
+    }
+    _taskNumber++;
+    _maybeScheduleTask();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _scheduler = SchedulerBinding.instance;
+    _controller = AnimationController(duration: Duration(milliseconds: 800), vsync: this);
+    final CurvedAnimation curve = CurvedAnimation(
+        parent: _controller,
+        curve: Interval(0.0, 1.0, curve: Curves.linear)
+    );
+    _flipAnimation = Tween(begin: 0.0, end: 1.0).animate(curve);
+    _controller.repeat();
+  }
+
+  dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Coin Flip Demo'),
+      ),
+      body: Padding(
+        padding: EdgeInsets.all(8.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          mainAxisAlignment: MainAxisAlignment.spaceAround,
+          children: <Widget>[
+            Column(
+              children: <Widget>[
+                _Coin(controller: _controller, flipAnimation: _flipAnimation),
+                SizedBox(height: 10.0),
+                _FlipButton(onPressed: _flipCoin)
+              ],
+            ),
+            _Results(heads: _headsCount, tails: _tailsCount, total: _totalFlipCount),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _Coin extends StatelessWidget {
+  _Coin({this.controller, this.flipAnimation});
+
+  final AnimationController controller;
+  final Animation flipAnimation;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.all(8.0),
+      child: AnimatedBuilder(
+        animation: controller,
+        builder: (BuildContext context, Widget child) {
+          return Center(
+            child: Container(
+              height: 150.0,
+              width: 150.0,
+              child: Transform(
+                transform: Matrix4.identity()
+                  ..rotateX(2 * pi * flipAnimation.value),
+                alignment: Alignment.center,
+                child: ClipOval(
+                    child: Container(
+                      color: Colors.amber,
+                    )
```

Comment:
add a trailing comma to make dartfmt do the right thing.

Reply:
Done.